File: src/ros_playground/src/tf_transform.cpp
#include <ros/ros.h>
#include <tf/transform_broadcaster.h>
#include <tf/transform_datatypes.h>

int main(int argc, char **argv) {
    ros::init(argc, argv, "tf_transform");
    ros::NodeHandle nh;
    ros::Rate rate(10.0);
    tf::TransformBroadcaster broadcaster;
    float angle = 0.0f;
    float x = 0.0f;
    float y = 0.0f;
    float z = 0.0f;

    ROS_INFO("tf_transform node started");
    while (nh.ok()) {
        x = 0.3 * cos(angle);
        y = 0.3 * sin(angle);
        z = 0.2;

        broadcaster.sendTransform(
            tf::StampedTransform(
                tf::Transform(tf::createQuaternionFromRPY(0.0, 0.0, angle), tf::Vector3(x, y, z)),
                ros::Time::now(), "tf1", "tf3"));
        broadcaster.sendTransform(
            tf::StampedTransform(
                tf::Transform(tf::Quaternion(0, 0, 0, 1), tf::Vector3(x, y, z*2)),
                ros::Time::now(), "tf1", "tf4"));

        rate.sleep();
        angle += 0.01;
    }
}

File: src/ros_playground/script/tf_transform.py
#! /usr/bin/env python3
import rospy
import tf
import math

def tf_transform():
    rospy.init_node('tf_transform', anonymous=False)
    tf_broadcaster = tf.TransformBroadcaster()
    rate = rospy.Rate(10.0)
    angle = 0.0  # rad
    rospy.loginfo('Start tf transform')
    while not rospy.is_shutdown():
        current_time = rospy.Time.now()
        x = math.cos(angle)*0.3
        y = math.sin(angle)*0.3
        z = 0.2
        quat = tf.transformations.quaternion_from_euler(0, 0, angle)
        tf_broadcaster.sendTransform((x, y, z), quat, current_time, 'tf3', 'tf1')
        tf_broadcaster.sendTransform((x, y, z*2), (0, 0, 0, 1), current_time, 'tf4', 'tf1')
        angle += 0.01
        rate.sleep()

if __name__ == '__main__':
    try:
        tf_transform()
    except rospy.ROSInterruptException:
        pass
